src/modules/module_23400.c

```diff
@@ -166,15 +167,15 @@ int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSE
 
   const u32 iter1 = hc_strtoul ((const char *) iter1_pos, NULL, 10);
 
-  if (iter1 <      1) return (PARSER_SALT_ITERATION);
-  if (iter1 > 999999) return (PARSER_SALT_ITERATION);
+  if (iter1 <     1) return (PARSER_SALT_ITERATION);
+  if (iter1 > UINT_MAX) return (PARSER_SALT_ITERATION);
```

Comment:
`iter1` is `u32` so condition `1 <= iter1 <= UINT_MAX` isn't just `iter != 0`?

Is it possible to store more than `UINT_MAX` in `iter1`?

```suggestion
  if (iter1 == 0) return (PARSER_SALT_ITERATION);
```